Vagrantfile

```diff
@@ -1,71 +1,302 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-clone_crystal_from_vagrant = lambda do |config|
+# Usage
+#
+# ```
+# $ vagrant up xenial64
+# $ vagrant ssh xenial64
+# vagrant@ubuntu-xenial:~$ cd ~/crystal/
+# vagrant@ubuntu-xenial:~$ make std_spec
+# ```
+#
+# The working directory is also available at /vagrant
+# but using the clone in ~/crystal helps isolating the binaries
+# from each platform and it can be faster on some circumstances
+# where files are shared with host since the clone is in the
+# native fs.
+#
+# The vagrant machines will be provisined with a native crystal
+# package or with the targz version downloaded from github.
+# Change INSTALL_GITHUB_TARGZ to choose the installation method.
+#
+# To use Makefile targets on some machines the `FLAGS=--no-debug`
+# might be needed to reduce the memory footprint.
+#
+# Notes:
+# * Running specs from /vagrant directly might cause some failures in
+#   specs related to permissions.
+# * bionic64 std_spec fails #6577
+# * xenial32 FLAGS=--no-debug is unable to run compiler_spec.
+# * trusty64 FLAGS=--no-debug is unable to build the compiler.
+# * precise32 is unable to build std_spec due to outdated libxml2. ~> 2.9.0 is required.
+# * precise32 will fail running std_spec related to reuse_port & ipv6.
+# * freebsd needs post install manual scripts in order to install crystal.
+# * alpine provisionning is failing due to openssl libressl issues.
+# * alpine is unable to to use the github .tar.gz compiler
+#
+
+INSTALL_GITHUB_TARGZ = false
```

Comment:
Couldn't this use an environment variable? This makes it easy to toggle source install without changing the Vagrantfile.

Reply:
done. I also added env vars for cpu and mem